addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2343,14 +2343,14 @@ var SnippetsMenu = Widget.extend({
                         await self._scrollToSnippet($target);
 
                         _.defer(async function () {
-                            self.trigger_up('snippet_dropped', {$target: $target});
                             self._disableUndroppableSnippets();
 
                             dragAndDropResolve();
 
                             await self._callForEachChildSnippet($target, function (editor, $snippet) {
                                 return editor.buildSnippet();
                             });
+                            self.trigger_up('snippet_dropped', {$target: $target});
```

Comment:
I am not confident with that change. Not only it changes a logic that devs may use in their own snippets (allowing `onBuilt` to be called on widgets-generated elements) but that means releasing the mutex (with `dragAndDropResolve()`) way before the animations are started and thus allowing edition while elements are being potentially added to the DOM.

I'd rather try to properly wait for animations/widgets to be completed (after the `snippet_dropped` trigger) like I explained in my previous message. It happens we thus have:

> 1. Drop the image gallery in the page
> 2. The gallery widget is created, removing the arrows
> 3. The gallery options are created, there is no arrow.

But (2.) is not properly awaited (should be something like `_refreshPublicWidgets` I guess).

Reply:
Ok. I wanted to make it consistent with the way that it is done when cloning a snippet,
Maybe the cloning should follow these steps in that order too : first starting the widgets, then creating the options ? 
For now we are properly awaiting the completion of each step, that's why the issue is not there when cloning an image gallery but it is done in the other way around, after cloning a snippet :

Reply:
Mmh, you are right, good point. Actually, about what I said yesterday about the logic of calling `onBuilt` on generated element... that's actually a bad logic as generated elements via cloning are not calling `onBuilt` afterwards anyway. Let's go back to what you did before, sorry: starting the widgets after they are built :+1: A bit more worried about the widgets start not being properly awaited but I guess we can ignore that for now, does not seem to hurt the cloning flow... :)